### ISSUE TITLE
User Guide organizational enhancments

### ISSUE THREAD

**Issue by cnpryer** [pola-rs/polars-book#160]
Successor to #153

tl;dr There's a lot of great content in Cookbook 1.0 which, if reorganized and updated, could provide a powerful revised flow for new and existing users reading through the book.

```mermaid
graph LR
  A[Intro] --> B[Getting Started] --> C[Fundamentals] --> D[Walkthrough] --> E[Recipes] --> F[References and Contributing]
```

## Motivation

If you're aware of what you're about to begin learning it's easier to spot where the fundamental information to learn is located. And from there you can branch out. For broader success we can revise the Cookbook to emphasize what should be emphasized (like Expressions) and refine the surrounding flow for the user to follow.

## Solution

| Step | Content     | Description |
| ---- | ----------- | ----------- |
| 1 | Intro  | An overview of the `Polars` `DataFrame` library. |
| 2 | Getting Started   | Create a `DataFrame` and get started. |
| 3 | Fundamentals         | Foundations and fundamental concepts for new and existing `Polars` users. |
| 4 | Walk-through   | A 5 minute walk-through of basic `Polars` API usage. |
| 5 | Recipes, References, and Contributing   | More material to refer to while learning and using `Polars`. |

To do this we can break up the work into a few PRs.

1. Update index tree structure to emphasize number 3 above and get the organization right (adds refined table above).
2. Add a fundamentals page leading into the existing expressions page.
3. Add #154 but as its own page.
4. Organize recipes and remaining updates.

Also the other topic I'd love to solve here would be the User Guide vs Cookbook confusion. IMO the book should be one or the other. If a third option *The Polars Book* could be considered that'd be sweet 😎.

The challenge we'll have with doing this is coordinating the cohesive revision of the book. If you're interested in coordinating with me on this please reach out 👋.

**Comment by cnpryer** [comment]
@braaannigan do you have any thoughts? I'm trying to avoid doing more of this stuff in a kind of vacuum.

**Comment by braaannigan** [comment]
@cnpryer Hi, no further thoughts beyond dropping the cookbook/recipes in favour of user guide terminology

**Comment by cnpryer** [comment]
To keep things fresh here, we should probably resolve #158 before any of this. From there I'll propose some possible plans for the fundamentals and walk-through improvements.

**Comment by cnpryer** [comment]
Now that #158 is resolved I can pick this up again. Also the cookbook was renamed so I can update some content here.
